src/__tests__/cli.ts

```diff
@@ -60,20 +60,28 @@ async function exec(
 
 async function createProject(
   files: Array<{ name: string; content: string }>,
+  directories: string[] = [],
+  baseDir = '.',
 ): Promise<string> {
   const randomId = Math.floor(Math.random() * 1000000);
 
   const testSpaceDir = path.join('.test-space', randomId.toString());
 
   await mkdir(testSpaceDir, { recursive: true });
 
+  await Promise.all(
+    directories.map((dir) =>
+      mkdir(path.join(testSpaceDir, dir), { recursive: true }),
+    ),
+  );
```

Comment:
I don't mind this, but I worry that the scenario setup could get fiddly as maintenance progresses in the future.  Having it pick up the directories from the file names using [`path.dirname`](https://nodejs.org/api/path.html#path_path_dirname_path) could make it a little bit more automatic and less to worry about without changing the implementation here too much. Something like:

```ts
  await Promise.all(
    files
      .map((file) => path.dirname(file.name))
      .map((dir) => mkdir(path.join(testSpaceDir, dir), { recursive: true })),
  );
  ```

Reply:
oh, didn't know about `path.dirname`
changing

Reply:
with this I can squash it with the other Promise.all

Reply:
done